src/components/InngestFunction.ts

```diff
@@ -251,7 +252,7 @@ export class InngestFunction<
      * add tools.
      */
     const fnArg = {
-      ...(data as { event: EventPayload }),
+      ...(data as { event: EventPayload; logger: ILogger }),
```

Comment:
```suggestion
      ...(data as { event: EventPayload; logger?: ILogger }),
```

Is `logger` optional?

Reply:
yup, nice catch.

Reply:
It'll always be there when they specify `logger` upon instantiation, right? If so, is there a way to make this optionality dependent on whether they specified `logger`?

Reply:
> It'll always be there when they specify logger upon instantiation, right?
Correct.

> If so, is there a way to make this optionality dependent on whether they specified logger?
That's some TS black magic I'm not aware of.
I can take a look. Hopefully the code doesn't result in something scary.

Reply:
This is the only article I found that seems to explain forcefully unwrapping optional arguments.
https://medium.com/terria/typescript-transforming-optional-properties-to-required-properties-that-may-be-undefined-7482cb4e1585

Definitely something I don't want to put my feet into, so here's a proposal.

If doing `logger?.info` seems too annoying, I can implement a dummy logger and make the type required instead of optional.
The dummy logger can just do `console.log` or something of the kind.

Reply:
Could use a generic. However, I needed to add a `// @ts-expect-error` because TS isn't smart enough to realize this is safe:

```ts
type Logger = {};

class InngestFunction<T extends Logger | undefined = undefined> {
  logger: T;

  constructor({ logger = undefined }: { logger?: T }) {
    // @ts-expect-error TS is unhappy because it's unable to recognize that
    // logger is only undefined when the constructor param is unspecified. In
    // other words, TS is trying to protect us against the impossible scenario
    // where we assign undefined to this.logger when the generic T is not
    // undefined.
    this.logger = logger;
  }
}

// a: {}
const a = new InngestFunction({ logger: {} }).logger;

// b: undefined
const b = new InngestFunction({ logger: undefined }).logger;

// c: undefined
const c = new InngestFunction({}).logger;
```

Reply:
> If doing logger?.info seems too annoying or non ergonomic, I can implement a dummy logger and make the type required instead of optional.
The dummy logger can just do console.log or something of the kind.

@darwin67 Yeah, I think it's reasonable to have a default `logger` that proxies `console` if nothing has been specified by the user.

Helps us avoid `logger?.*` both for the user and the fun type inference above we'd have to do to definitively say it either does or doesn't exist within the function body.

Reply:
Default/dummy logger sounds best 👍